upload-api/functions/validate-email.jsx

```diff
@@ -0,0 +1,133 @@
+import * as Sentry from '@sentry/serverless'
+import { authorize } from '@web3-storage/upload-api/validate'
+import { getServiceSigner } from '../config.js'
+import { Email } from '../email.js'
+import { createDelegationsTable } from '../tables/delegations.js'
+import { createDelegationsStore } from '../buckets/delegations-store.js'
+import { createInvocationStore } from '../buckets/invocation-store.js'
+import { createWorkflowStore } from '../buckets/workflow-store.js'
+import { createSubscriptionTable } from '../tables/subscription.js'
+import { createConsumerTable } from '../tables/consumer.js'
+
+import { useProvisionStore } from '../stores/provisions.js'
+import {
+  HtmlResponse,
+  ValidateEmail,
+  ValidateEmailError,
+  PendingValidateEmail,
+} from '../html.jsx'
+
+Sentry.AWSLambda.init({
+  environment: process.env.SST_STAGE,
+  dsn: process.env.SENTRY_DSN,
+  tracesSampleRate: 1.0,
+})
+
+/**
+ * @param {HtmlResponse} response
+ */
+export function toLambdaResponse (response) {
+  const { status = 200, headers, body } = response
+  return {
+    statusCode: status,
+    headers,
+    body: body && Buffer.from(response.getStringBody()).toString('base64'),
+    isBase64Encoded: true
+  }
+}
+
+/**
+ * AWS HTTP Gateway handler for GET /validate-email
+ *
+ * @param {import('aws-lambda').APIGatewayProxyEventV2} request
+ */
+export async function validateEmailGet (request) {
+  if (!request.queryStringParameters?.ucan) {
+    return toLambdaResponse(new HtmlResponse(
+      <ValidateEmailError msg={'Missing delegation in the URL.'} />
+    ))  
+  }
+
+  return toLambdaResponse(new HtmlResponse(<PendingValidateEmail autoApprove={true} />))
+}
+
+export const preValidateEmail = Sentry.AWSLambda.wrapHandler(validateEmailGet)
+
+function createAuthorizeContext () {
+  const {
+    ACCESS_SERVICE_URL = '',
+    ACCESS_SERVICE_DID = '',
+    AWS_REGION = '',
+    DELEGATION_TABLE_NAME = '',
+    R2_DELEGATION_BUCKET_ENDPOINT = '',
+    R2_DELEGATION_BUCKET_ACCESS_KEY_ID = '',
+    R2_DELEGATION_BUCKET_SECRET_ACCESS_KEY = '',
+    R2_DELEGATION_BUCKET_NAME = '',
+    INVOCATION_BUCKET_NAME = '',
+    WORKFLOW_BUCKET_NAME = '',
+    POSTMARK_TOKEN = '',
+    PRIVATE_KEY = '',
```

Comment:
PRIVATE_KEY is a secret and not env. SST has something for it :) See https://github.com/web3-storage/w3infra/blob/main/upload-api/functions/ucan-invocation-router.js#L96